modules/redis/redis.go

```diff
@@ -27,10 +40,33 @@ const (
 
 type RedisContainer struct {
 	testcontainers.Container
+	settings options
 }
 
+// ConnectionString returns the connection string for the Redis container.
+// It uses the default 6379 port.
 func (c *RedisContainer) ConnectionString(ctx context.Context) (string, error) {
-	mappedPort, err := c.MappedPort(ctx, "6379/tcp")
+	return c.connectionString(ctx, nat.Port(redisPort))
+}
+
+// ConnectionStringTLS returns the connection string for the Redis container using TLS.
+// It returns an error if TLS is not enabled, else the TLS port defined in the options
+// is used to build the connection string.
+func (c *RedisContainer) ConnectionStringTLS(ctx context.Context) (string, error) {
```

Comment:
suggestion: eliminate as its either TLS or not see comment about [port = 0 when tls is enabled](https://raw.githubusercontent.com/redis/redis/unstable/redis.conf).